src/client/math/matrix2.ts

```diff
@@ -0,0 +1,79 @@
+import { action } from 'mobx'
+import { computed } from 'mobx'
+import { observable } from 'mobx'
+import { Vector2 } from './vector2'
+
+export class Matrix2 {
+  @observable public x: Vector2
+  @observable public y: Vector2
+
+  public constructor(x: Vector2, y: Vector2) {
+    this.x = x
+    this.y = y
+  }
+
+  public static of() {
+    return new Matrix2(new Vector2(0, 0), new Vector2(0, 0))
+  }
+
+  public static from(mat?: {
+    x?: { x?: number, y?: number },
+    y?: { x?: number, y?: number }
+  } | null): Matrix2 {
+    if (!mat) {
```

Comment:
I'm pretty hesitant of this functionality here (I realise its for consistency with Vector2). I'm not sure which situations we would expect to provide a `null` object, but they _should_ be rare if at all. I think I'd prefer a type error instead. i.e. I think we should just make everything required instead by removing the `?` and this conditional.